common-npm-packages/artifacts-common/credentialProviderUtils.ts

```diff
@@ -119,36 +119,54 @@ export async function configureCredProvider(protocol: ProtocolType, serviceConne
 }
 
 /**
- * Configure the credential provider to provide credentials for feeds within the pipeline's organization,
- * as well as for a single Entra backed service connection.
+ * Configure credentials for the given feed using the provided 'Azure Devops' service connection.
+ * Only external feeds are supported, will throw if the feed provided is internal.
  */
-export async function configureEntraCredProvider(protocol: ProtocolType, feedUrl: string, entraWifServiceConnectionName: string) {
-    await configureCredProviderForSameOrganizationFeeds(protocol);
+export async function configureEntraCredProvider(protocol: ProtocolType, entraWifServiceConnectionName: string, feedUrl: string | undefined) : Promise<boolean> {
+    const connectionData = await getConnectionDataForProtocol(protocol);
+    const packagingAccessMappings = getPackagingAccessMappings(connectionData.locationServiceData)
+    const allPrefixes: string[] = [...new Set(packagingAccessMappings.map(prefix => prefix.uri))];
+    const isFeedInternal = allPrefixes.some(prefix => feedUrl.startsWith(prefix));
+
+    if (isFeedInternal) {
+        throw new Error(tl.loc("Error_InternalFeedsNotSupported"));
+    }
 
     const feedTenant = await getFeedTenantId(feedUrl);
     let token = await getFederatedWorkloadIdentityCredentials(entraWifServiceConnectionName, feedTenant);
     if (token) {
         configureCredProviderForServiceConnectionFeeds([new EntraServiceConnection({uri:feedUrl}, entraWifServiceConnectionName, token)]);
```

Comment:
does this ever return false? or is it only throwing an error in the false case?

Reply:
Hmm it only throws an error and doesn't return false, only true to show that it was successful. I suppose I can use the throw to exit early and remove the use of the boolean

Reply:
I typically prefer booleans to void functions as well, but here I think it might be awkward because the `configureCredProviderForServiceConnectionFeeds` is also void, and there we are not checking if that one is successful. So in the event that getting the token was successful, but `configureCredProviderForServiceConnectionFeeds` silently fails, the returned true here kinda gives off a false sense of success.